src/server/pipeline.py

```diff
@@ -241,10 +241,13 @@ async def perform_handshake(
                 f" established"
             )
         )
+
         # Only yield a connection if the handshake is succesful and the connection is not a duplicate.
-        yield connection, global_connections
+        async for conn in global_connections.successful_handshake(connection):
```

Comment:
Why add a generator here?

Reply:
if it's a feeler connection, it'll be an empty generator. same intend, to move case handling from pipeline to connection.